linkml/generators/rustgen/rustgen.py

```diff
@@ -130,20 +143,38 @@ def get_key_or_identifier_slot(cls: ClassDefinition, sv: SchemaView) -> Optional
             return slot
     return None
 
-def get_inline_mode(s: SlotDefinition, sv: SchemaView) -> SlotInlineMode:
+
+def get_identifier_slot(cls: ClassDefinition, sv: SchemaView) -> Optional[SlotDefinition]:
+    induced_slots = sv.class_induced_slots(cls.name)
+    for slot in induced_slots:
+        if slot.identifier:
+            return slot
+    return None
+
+
+def get_inline_mode(s: SlotDefinition, sv: SchemaView) -> tuple[SlotContainerMode, SlotInlineMode]:
```

Comment:
wow this sure is complicated (not your fault) - definitely seems like something that should be in a more generic place, since i am sure that other generators all reimplement this exact same thing

Reply:
agree. i'll add some test for this one later too (and find a better name for it, its not only about inline mode).